cmd/trust/sudi.go

```diff
@@ -3,122 +3,106 @@ package main
 import (
 	"crypto/x509"
 	"crypto/x509/pkix"
-	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
 	"time"
 
-	"github.com/apex/log"
 	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/project-machine/trust/pkg/trust"
 	"github.com/urfave/cli"
 )
 
-var genSudiCmd = cli.Command{
-	Name:      "gen-sudi",
+var sudiCmd = cli.Command{
+	Name:      "sudi",
 	Usage:     "Generate and sign sudi cert",
 	UsageText: "CACert, private-key, output-dir, product-uuid [, machine-uuid]",
-	Action:    doGenSudi,
+	Subcommands: []cli.Command{
+		cli.Command{
+			Name:      "list",
+			Action:    doListSudi,
+			Usage:     "list sudi keys",
+			ArgsUsage: "<keyset-name> <project-name>",
+		},
+		cli.Command{
+			Name:      "add",
+			Action:    doGenSudi,
+			Usage:     "add a new sudi key to project",
+			Flags: []cli.Flag{
+				cli.StringFlag {
+					Name: "uuid",
+					Usage: "specify a machine-uuid.  If unspecified, use random.",
```

Comment:
maybe we could consider for this Usage message, "Usage: "specify a uuid to be used for the serial-number. If not specified, one will be randomly assigned."

Reply:
Oh, I think I have this confuddled - for sudi, the uuid would *be* the serial number, right?

So we either specify a serial number - like SN001 or WZP250210JA - or some uuid - or else trust creates a UUID, and then machine is expected ot specify that uuid as the /sys/class/dmi/id/board_serial?